ui/app/components/totp-edit.js

```diff
@@ -7,48 +7,55 @@ import Component from '@glimmer/component';
 import { action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
 import { service } from '@ember/service';
-import { isBlank } from '@ember/utils';
+import { task } from 'ember-concurrency';
+import { waitFor } from '@ember/test-waiters';
 
-// TODO add jsdoc documentation
+/**
+ * @module TotpEdit
+ * `TotpEdit` is a component that allows you to create, view or delete a TOTP key or view the QR code of the key.
```

Comment:
Since we were confused about this earlier this week, adding the generate functionality might be helpful here
```suggestion
 * `TotpEdit` is a component that allows you to create, view or delete a TOTP key. When creating a key if `generate` and `exported` are true then after a successful save the UI renders a QR code for the generated key.
```

Reply:
Good idea!